src/napari/_qt/_tests/test_viewer.py

```diff
@@ -0,0 +1,93 @@
+"""Tests of the Viewer class that interact directly with the Qt code"""
+
+import textwrap
+from unittest.mock import MagicMock
+
+import pytest
+from qtpy.QtCore import QUrl
+
+
+@pytest.mark.usefixtures('builtins')
+def test_drop_python_file(make_napari_viewer, tmp_path):
+    """Test that dropping a python file on the viewer works."""
+    viewer = make_napari_viewer()
+    filename = tmp_path / 'test.py'
+    file_content = textwrap.dedent("""
+    import numpy as np
+    from napari import Viewer
+
+    data = np.zeros((10, 10))
+    viewer = Viewer()
+    viewer.add_image(data, name='Dropped Image')
+    """)
+    filename.write_text(file_content)
+
+    # Simulate dropping the file
+    mock_event = MagicMock()
+    mock_event.mimeData.return_value.urls.return_value = [
+        QUrl(filename.as_uri())
+    ]
+    viewer.window._qt_viewer.dropEvent(mock_event)
+
+    # Check that the file was executed
+    assert viewer.layers[0].name == 'Dropped Image'
+
+
+@pytest.mark.usefixtures('builtins')
+def test_drop_python_file_3d(make_napari_viewer, tmp_path):
+    """Test that dropping a python file on the viewer works."""
+    viewer = make_napari_viewer()
+    filename = tmp_path / 'test.py'
+    file_content = textwrap.dedent("""
+    import numpy as np
+    from napari import Viewer
+
+    data = np.zeros((2, 10, 10))
+    viewer = Viewer(ndisplay=3)
+    viewer.add_image(data, name='Dropped Image')
+    """)
+    filename.write_text(file_content)
+
+    # Simulate dropping the file
+    mock_event = MagicMock()
+    mock_event.mimeData.return_value.urls.return_value = [
+        QUrl(filename.as_uri())
+    ]
+    viewer.window._qt_viewer.dropEvent(mock_event)
+
+    # Check that the file was executed
+    assert viewer.layers[0].name == 'Dropped Image'
+    assert viewer.dims.ndim == 3
+
+
+@pytest.mark.usefixtures('builtins')
+def test_drop_python_file_double_viewer(make_napari_viewer, tmp_path):
```

Comment:
This test is confusing. Can we modify this test to reflect the purpose of testing two viewers? Existing viewer (1) and new viewer (2)?

Reply:
It is exactly what this test does. You mean to update the description?